### PR TITLE
DRAFT:ENH: interpolate.AAA: add integrate method

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

follow up to https://github.com/scipy/scipy/pull/21167
#### What does this implement/fix?

Adds a prototype `integrate` method to AAA based on direct rational quadrature from https://www.sciencedirect.com/science/article/pii/S0377042713001714. Clearly, the interface and final implementation are unfinished and still need further testing!

#### Additional information

@mdhaber I thought I would try this out as you expressed an integration in the original review, is this suitable for your use case and would you be interested in testing this out? Also, I know there were some concerns about if array API support could be added to this, I think the core algorithm could be converted as it is just the pole/roots functions that need to solve a generalised eigenvalue problem. EDIT: Looking at the original review not sure this is quite what you were after but would be nice to add regardless.

### CODE REVIEW

File: scipy/interpolate/_aaa.py [hunk 1]
```diff
@@ -416,3 +417,13 @@ def roots(self):
...
             self._roots = zer[np.isfinite(zer)]
         return self._roots
+
+    def integrate(self, a, b):
+        def quad_weight_intergrand(x, weight, support_point):
+            return (weight/(x - support_point))/np.sum(self.weights/(x-self.support_points))
+        
+        weights = np.array(
+            [scipy.integrate.quad(quad_weight_intergrand, a, b, args=(i, j))[0]
+             for i, j in zip(self.weights, self.support_points)]
```
Comment (j-bowhay) [inline scipy/interpolate/_aaa.py:427]: This could be replaced with `_tanhsinh`

**Comment by mdhaber** [comment]
Sure, I can review this. 
Yeah, I was looking for an analytical solution like for splines, but this is still good to have.

File: scipy/interpolate/_aaa.py [hunk 2]
```diff
@@ -416,3 +417,13 @@ def roots(self):
             zer = scipy.linalg.eigvals(E, B)
             self._roots = zer[np.isfinite(zer)]
         return self._roots
+
+    def integrate(self, a, b):
+        def quad_weight_intergrand(x, weight, support_point):
```
Comment (mdhaber) [inline scipy/interpolate/_aaa.py:422]:
```suggestion
        def quad_weight_integrand(x, weight, support_point):
```

**Comment by j-bowhay** [comment]
> Sure, I can review this. Yeah, I was looking for an analytical solution like for splines, but this is still good to have.

Actually, it might be worth holding fire on this, there may be a better way of doing this!

**Pull request closed without merging**